src/cmd_kv.cc

```diff
@@ -216,6 +216,37 @@ void BitCountCmd::DoCmd(PClient* client) {
   client->AppendInteger(static_cast<int64_t>(count));
 }
 
+IncrCmd::IncrCmd(const std::string& name, int16_t arity)
+    : BaseCmd(name, arity, CmdFlagsReadonly, AclCategoryRead | AclCategoryString) {}
+
+bool IncrCmd::DoInitial(pikiwidb::PClient* client) {
+  client->SetKey(client->argv_[1]);
+  return true;
+}
+
+void IncrCmd::DoCmd(pikiwidb::PClient* client) {
+  PObject* value = nullptr;
+  PError err = PSTORE.GetValueByType(client->Key(), value, PType_string);
+  if (err == PError_notExist) {
+    value = PSTORE.SetValue(client->Key(), PObject::CreateString(0));
+  }
+
+  if (err != PError_ok) {
+    client->SetRes(CmdRes::kErrOther);
+    return;
+  }
+
+  if (value->encoding != PEncode_int) {
+    client->SetRes(CmdRes::kInvalidInt);
+    return;
+  }
+
+  intptr_t oldVal = static_cast<intptr_t>(reinterpret_cast<std::intptr_t>(value->value));
+  value->Reset(reinterpret_cast<void*>(static_cast<std::intptr_t>(oldVal + 1)));
```

Comment:
```suggestion
  value->Reset(reinterpret_cast<void*>(oldVal + 1));
```
这个`static_cast`是不必要的